### ISSUE TITLE
emmean support for monotonic effects

### ISSUE THREAD

**Issue by mattansb** [paul-buerkner/brms#1134]
`emmeans` seems unable to find monotonic effects (missing from the reference grid):

```r
library(brms)
library(emmeans)

income_options <- c("below_20", "20_to_40", "40_to_100", "greater_100")
income <- factor(sample(income_options, 100, TRUE), 
                 levels = income_options, ordered = TRUE)
mean_ls <- c(30, 60, 70, 75)
ls <- mean_ls[income] + rnorm(100, sd = 7)
dat <- data.frame(income, ls)

fit1 <- brm(ls ~ mo(income), data = dat)


emmeans(fit1,~income)
#> Error in emmeans(fit1, ~income) : 
#>   No variable named income in the reference grid

ref_grid(fit1)
#> 'emmGrid' object with variables:
#>   1 = 1
```

**Comment by paul-buerkner** [comment]
Yes that is true. I am not yet sure, how we shall best handle monotonic effects in emmeans.

**Comment by mattansb** [comment]
I think this would be handled best if it were in the reference grid as a "factor".

**Comment by mattansb** [comment]
This issue also seems to be solved by the suggestion I made here: https://github.com/paul-buerkner/brms/issues/907#issuecomment-856457851

```R
library(brms)
library(emmeans)

income_options <- c("below_20", "20_to_40", "40_to_100", "greater_100")
income <- factor(sample(income_options, 100, TRUE), 
                 levels = income_options, ordered = TRUE)
mean_ls <- c(30, 60, 70, 75)
ls <- mean_ls[income] + rnorm(100, sd = 7)
dat <- data.frame(income, ls)

fit1 <- brm(ls ~ mo(income), data = dat)


emmeans(fit1,  ~ income)
#>  income      emmean lower.HPD upper.HPD
#>  below_20      30.3      27.4      33.2
#>  20_to_40      60.1      57.0      63.3
#>  40_to_100     69.4      66.3      72.6
#>  greater_100   75.2      72.7      77.8
#> 
#> Point estimate displayed: median 
#> HPD interval probability: 0.95 
```

**Comment by paul-buerkner** [comment]
Implemented via #1177
